android/api.md

```diff
@@ -594,6 +609,252 @@ private static String MyOIDCAuthProvider(){
+                                   .useClientDatabasePrefix(true)
+                                   .cognitoUserPoolsAuthProvider(new CognitoUserPoolsAuthProvider() {
+									            @Override
+									            public String getLatestAuthToken() {
+									                try {
+									                    String idToken = AWSMobileClient.getInstance().getTokens().getIdToken().getTokenString();
+									                    return idToken;
+									                } catch (Exception e) {
+									                    e.printStackTrace();
+									                    return null;
+									                }
+									            }
+									        })
+                                   .build();
+    }
+}
+```
+
+This is what the usage would look like.
+
+```java
+ClientFactory.getAppSyncClient(AppSyncClientMode.PRIVATE).query(fooQuery).execute();
+```
+
+The following example uses `API_KEY` as the default authorization mode and `AWS_IAM` as an additional authorization mode.
+
+```
+type Post @aws_api_key
+          @aws_iam {
+	id: ID!
+	author: String!
+	title: String
+	content: String
+	url: String @aws_iam
+	ups: Int @aws_iam
+	downs: Int @aws_iam
+	version: Int!
+}
+```
+
+1. Add a post (Mutation) through `CLIENTS[PRIVATE]` using `AWS_IAM ` authorization mode.
+
+2. Query the post through `CLIENTS[PRIVATE]` using `AWS_IAM` authorization mode.
+
+3. Query the post through `CLIENTS[PUBLIC]` using `API_KEY` authorization mode
+
+```java
+apiKeyAppSyncClient.query(GetPostQuery.builder().id(id).build())
+        .responseFetcher(responseFetcher)
+        .enqueue(new GraphQLCall.Callback<GetPostQuery.Data>() {
+            @Override
+            public void onResponse(@Nonnull Response<GetPostQuery.Data> response) {
+                GetPost post = response.data().getPost();
+                post.id();
+                post.author();
+                post.title();
+                post.content();
+                post.version();
+                
+                post.url(); // Null - because it's not authorized
+                post.ups(); // Null - because it's not authorized
+                post.downs(); // Null - because it's not authorized
+                
+                if (response.hasErrors()) {
+	                	Log.d(TAG, response.errors().get(0).message()); // "Not Authorized to access url on type Post"
+	                	Log.d(TAG, response.errors().get(1).message()); // "Not Authorized to access ups on type Post"
+	                	Log.d(TAG, response.errors().get(2).message()); // "Not Authorized to access downs on type Post"
+                }
+            }
+
+            @Override
+            public void onFailure(@Nonnull ApolloException e) {
+                e.printStackTrace();
+            }
+        });
+```
+
+**Code based client instantiation**
+
+Alternative to using `awsconfiguration.json`, you could use the `AWSAppSyncClientBuilder`.
```

Comment:
"Alternative to using `awsconfiguration.json`, you could use the `AWSAppSyncClientBuilder`." --> "You can use the `AWSAppSyncClientBuilder` as an alternative to using 'awsconfiguration.json'"

Reply:
Changed.